src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerOutputFormatter.cs

```diff
@@ -183,11 +185,31 @@ public override Task WriteResponseBodyAsync([NotNull] OutputFormatterContext con
                     obj = wrapperProvider.Wrap(obj);
                 }
 
-                var dataContractSerializer = CreateSerializer(wrappingType);
+                var dataContractSerializer = GetCachedSerializer(wrappingType);
                 dataContractSerializer.WriteObject(xmlWriter, obj);
             }
 
             return Task.FromResult(true);
         }
+
+        private DataContractSerializer GetCachedSerializer(Type type)
+        {
+            if (type == null)
```

Comment:
`TryGetValue` produces an exception when `null` is passed. We have some tests that pass null to `CanRead` and expect `false`.
Eg. https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNet.Mvc.Xml.Test/XmlSerializerInputFormatterTest.cs#L50